.github/workflows/trivy-scan.yml

```diff
@@ -0,0 +1,126 @@
+name: Trivy CVE Dependency Scanner
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+jobs:
+  scan-latest-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get Latest Release Docker Image Sha
+        id: latest-sha
+        run: |
+          # Get the latest released docker image sha
+          curl -sL https://api.github.com/repos/vmware-tanzu/carvel-secretgen-controller/releases/latest | jq -r '.assets[].browser_download_url' | wget -i -
+
+          echo ::set-output name=image::$(yq eval '.spec.template.spec.containers[0].image' release.yml -N -j | jq 'select(. != null)' -r)
+          echo ::set-output name=tag::$(curl -sL https://api.github.com/repos/vmware-tanzu/carvel-secretgen-controller/releases/latest | jq -r '.tag_name')
+      - name: Install Trivy
+        run: |
+          brew install aquasecurity/trivy/trivy
+      - name: Run Trivy
+        run: |
+          trivy image ${{ steps.latest-sha.outputs.image }}
+          trivy image --format json --output trivy-results-image-latest.json ${{ steps.latest-sha.outputs.image }}
+      - name: Check for new Vulnerabilities
+        run: |
+          set -o pipefail
+
+          summary="Trivy scan has found new vulnerabilities in ${{steps.latest-sha.outputs.tag}} check https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}"
+
+          vulnCount=$(jq '[ .Results[].Vulnerabilities ] | length' trivy-results-image-latest.json)
+          if [[ $vulnCount -eq 0 ]]; then
+            summary="Trivy Scan has not found any new Security Issues in ${{steps.latest-sha.outputs.tag}}"
+          fi
+
+          echo "SUMMARY=$summary" >> $GITHUB_ENV
+      - name: Send Slack Notification
+        if: success()
+        uses: slackapi/slack-github-action@v1.15.0
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+        with:
+          channel-id: C02MTA7F2G1
+          slack-message: "${{ env.SUMMARY }}"
+      - name: Send Failure notification
+        if: failure()
+        uses: slackapi/slack-github-action@v1.15.0
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+        with:
+          channel-id: G01FTP43JMQ
+          slack-message: "Trivy scan workflow [${{steps.latest-sha.outputs.tag}}] failed. Please check the latest github action run for trivy scanner."
+  scan-develop-branch:
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v1
+        with:
+          go-version: "1.17"
+      - name: Build the secretgen-controller artifacts
+        run: |
+          ./hack/install-deps.sh
+          ./hack/build.sh
+
+          # docker image
+          docker build -t docker.io/carvel/secretgen-controller:${{ github.sha }} .
```

Comment:
I was a little surprised we're (still?) using docker.io/carvel --  was there a conversation in KC about whether to use docker.ui vs ghcr.io ?

Reply:
I don't think we need either `docker.io` or `ghcr.io` since this is just a local build anyway. Not sure if trivy cares about this or not.

Reply:
Yep, as daniel says. It's not pushing to an external registry, just locally. I don't know if this can removed, I didn't try... the trivy example had it this way, so I used that. @joe-kimmel-vmw